### ISSUE TITLE
Tuple length inference does not work with list argument

### ISSUE THREAD

**Issue by nevercast** [microsoft/pyright#1034]
**Describe the bug**
When a tuple is constructed from a sequence (in this example, a list), Pyright sees the tuple length as 1 which causes `Tuple size mismatch` errors later in the source. 

**To Reproduce**
I am using Pyright with VSCode on Ubuntu 18.04, and writing Python code without using `typings` or type hint comments/annotations. Types are inferred.
![image](https://user-images.githubusercontent.com/834342/93147259-d1ec9100-f744-11ea-97cf-a482b77d76a0.png)
![image](https://user-images.githubusercontent.com/834342/93147280-e16bda00-f744-11ea-9a43-0f64e647db71.png)

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/834342/93147007-6a364600-f744-11ea-9abc-d7459ab669d8.png)
![image](https://user-images.githubusercontent.com/834342/93147659-ee3cfd80-f745-11ea-87b2-fc4b726679bc.png)

```python
bla = 'left|right'
if '|' in bla:
    left, right = tuple(bla.split('|'))
```

**Expected behavior**

The ideal behaviour is that `bla.split` has a minimum length of 2 because `if '|' in bla` asserts that is the case. Acceptable behaviour would be that the right hand side has an unknown length and no error is raised. Foregoing these two behaviours, a fallback to specify the minimum length of the tuple or list with type annotations so that further errors are avoided.

![image](https://user-images.githubusercontent.com/834342/93147739-27756d80-f746-11ea-884e-3c929bbe580f.png)

**Comment by erictraut** [comment]
The `tuple` class is defined in `builtins.pyi` with the following constructor:
```python
class tuple(Sequence[_T_co], Generic[_T_co]):
    def __new__(cls: Type[_T], iterable: Iterable[_T_co] = ...) -> _T: ...
```

Applying the normal type-checking rules for this constructor yields the type `tuple[str]` in your example, which is why you're seeing the error.

It appears that mypy special-cases the tuple constructor, transforming the resulting type from `tuple[_T_co]` into `tuple[_T_co, ...]`. I've replicated this special-case logic in Pyright, which will provide the behavior you're looking for.

In general, I don't like adding special-case logic like this for specific types because it makes for unmaintainable code and an unscalable solution, but tuples demand some special cases like this because they're one of few types in the Python type system that is variadic (i.e. supports an arbitrary number of type arguments). So in this case, I'll make an exception to my general rule against special-case handling.

**Comment by nevercast** [comment]
Thank you very much for going against the grain on this to provide special-case behaviour. Will look forward to this code no longer flagging an error in the next Pyright version. :tada:

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.72, which I just published. It will also be in the next release of Pylance.
